cpp/src/gandiva/tests/micro_benchmarks.cc

```diff
@@ -213,4 +214,82 @@ TEST_F(TestBenchmarks, TimedTestFilterLike) {
   EXPECT_LE(elapsed_millis, 600 * tolerance_ratio);
 }
 
+// following two tests are for benchmark optimization of
+// in expr. will be used in follow-up PRs to optimize in expr.
+
+TEST_F(TestBenchmarks, TimedTestMultiOr) {
```

Comment:
Wow, we shouldn't put benchmarks in unit tests. We use separate gbenchmark-based executables for that. You can look for examples in the Arrow source tree.

Reply:
agreed, we've thinking about it..created ARROW-3609 for the same.